lib/dartino/launch_dartino.dart

```diff
@@ -0,0 +1,121 @@
+import 'dart:async';
+
+import 'package:atom/node/fs.dart';
+import 'package:atom/node/process.dart';
+import 'package:logging/logging.dart';
+
+import '../atom.dart';
+import '../launch/launch.dart';
+import '../projects.dart';
+import '../state.dart';
+
+final Logger _logger = new Logger('atom.dartino_launch');
+
+class DartinoLaunchType extends LaunchType {
+  static void register(LaunchManager manager) =>
+      manager.registerLaunchType(new DartinoLaunchType());
+
+  DartinoLaunch _lastLaunch;
+
+  DartinoLaunchType() : super('dartino');
+
+  bool canLaunch(String path) {
+    DartProject project = projectManager.getProjectFor(path);
+    if (project == null || !project.isDartinoProject()) return false;
+    if (!analysisServer.isExecutable(path)) return false;
+    return true;
+  }
+
+  List<String> getLaunchablesFor(DartProject project) {
+    if (project == null || !project.isDartinoProject()) return [];
+    var files = <String>[];
+    visit(Directory dir) {
+      for (Entry entry in dir.getEntriesSync()) {
+        if (entry.isDirectory()) {
+          visit(entry);
+          continue;
+        }
+        var path = entry.getPath();
+        if (path != null && path.endsWith('dart')) files.add(path);
+      }
+    }
+    visit(project.directory);
+    return files;
+    //TODO(danrubel) add Dartino launch support to analysis server
+    // return analysisServer
+    //     .getExecutablesFor(project.path)
+    //     .where((String path) => path.endsWith('dart'))
+    //     .toList();
+  }
+
+  Future<Launch> performLaunch(
+      LaunchManager manager, LaunchConfiguration configuration) async {
+    String path = configuration.primaryResource;
+
+    DartProject project = projectManager.getProjectFor(path);
+    if (project == null || !project.isDartinoProject()) {
+      throw "File not in a Dartino project.";
+    }
+
+    // Sdk sdk = dartino.sdkFor(project.directory);
+    // if (sdk == null) {
+    //   throw 'No SDK found for $project';
+    // }
+
+    await _killLastLaunch();
+    _lastLaunch = new DartinoLaunch(manager, this, configuration);
+    manager.addLaunch(_lastLaunch);
+    // TODO (danrubel) to be implemented
+    // sdk.launch(_lastLaunch);
+    atom.notifications.addError('Not implemented yet');
+    return _lastLaunch;
+  }
+
+  String getDefaultConfigText() {
+    return '';
```

Comment:
You can put default option values here that the user can edit. Things like checked mode, debug, cli args, ...